XenAdmin/Controls/Ballooning/VMMemoryControlsBasic.cs

```diff
@@ -100,21 +100,15 @@ protected override void OnPaint(PaintEventArgs e)
                     }
                     if (identical)
                     {
-                        switch (vm0.GetVirtualisationStatus)
-                        {
-                            case VM.VirtualisationStatus.OPTIMIZED:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_NOTSUPPORTED_PLURAL;
-                                break;
-                            case VM.VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_NOTOOLS_PLURAL;
-                                break;
-                            case VM.VirtualisationStatus.PV_DRIVERS_OUT_OF_DATE:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_OLDTOOLS_PLURAL;
-                                break;
-                            default:  // VM.VirtualisationStatus.UNKNOWN
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_VMS;
-                                break;
-                        }
+                        var status = vm0.GetVirtualisationStatus;
+                        if (status.HasFlag(VM.VirtualisationStatus.IO_DRIVERS_INSTALLED | VM.VirtualisationStatus.MANAGEMENT_INSTALLED))

```

Comment:
Agreed, I changed them to use && to make it more explicit.

In this file: We do not need both, but this is the method that is trying to guess what has gone wrong, so it can be a bit different - I think. From this point of view I would leave this basically as is (after a small change I have made). It will suggest to install I/O drivers on NewVMs when failed. (More specifically, XenBus is needed, but having seen the presence of IO drivers or Management agent implies we have that. Although as this is an error message, I decided to refer to missing IO drivers only - for new VMs)
For CanExecute we check the feature-balloon - that is done the right way.

In other files I made some more changes, please find them in my last commit.